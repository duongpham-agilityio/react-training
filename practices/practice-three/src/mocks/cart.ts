import { TCartItemProps } from '@/components/Cart/components';

export const cartItemProps: TCartItemProps = {
  productId: 1,
  description: 'Adidas Menâ€™s T-Shirt',
  name: 'Adidas T-Shirt',
  imageURL:
    'https://static.nike.com/a/images/t_PDP_1280_v1/f_auto,q_auto:eco/d455e5b6-23e9-4f06-8b86-32610d838937/renew-run-3-road-running-shoes-395S7c.png',
  price: 200,
  quantity: 2,
  onChangeQuantity: () => {},
  onRemove: () => {},
};
