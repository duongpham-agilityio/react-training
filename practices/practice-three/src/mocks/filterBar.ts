// Types
import { FilterBarProps } from '@/components';

export interface IFilterOption {
  name: string;
  value: string;
}

export const filterOptions: IFilterOption[] = [
  {
    name: 'All',
    value: 'all',
  },
  {
    name: 'Fashion',
    value: 'fashion',
  },
  {
    name: 'Electrics',
    value: 'electrics',
  },
  {
    name: 'Books',
    value: 'books',
  },
];

export const filterBarProps: FilterBarProps = {
  value: '',
  onChange: () => {},
};
