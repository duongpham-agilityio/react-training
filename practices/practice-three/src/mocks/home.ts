import { IProduct } from '@/interface';

export const products: IProduct[] = [
  {
    name: 'Tasty Plastic Mouse',
    description: 'Quos tempora adipisci quidem ipsum.',
    imageURL: 'https://loremflickr.com/640/480/fashion',
    price: 430,
    quantity: 10,
    isLiked: true,
    id: 1,
    category: 'fashion',
    createdAt: new Date(),
  },
  {
    name: 'Tasty Wooden Chair',
    description:
      'Ab repellat ullam autem. Esse facere facere debitis dicta quidem nihil. Saepe suscipit ex.',
    imageURL: 'https://loremflickr.com/640/480/fashion',
    price: 430,
    quantity: 10,
    isLiked: false,
    id: 2,
    category: 'fashion',
    createdAt: new Date(),
  },
];
