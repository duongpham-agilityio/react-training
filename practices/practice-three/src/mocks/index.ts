export * from './productCard';
export * from './filterBar';
export * from './inputForm';
export * from './pagination';
