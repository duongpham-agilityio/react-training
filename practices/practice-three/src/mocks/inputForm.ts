// Components
import { TInputFormProps } from '@/components/common';

export const inputFormProps: TInputFormProps = {
  label: 'Email',
  value: '',
  placeholder: 'Please enter email',
  onChange: () => {},
};
