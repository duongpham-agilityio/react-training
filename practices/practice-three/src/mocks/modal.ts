import { ModalProps } from '@/components';

export const defaultModalProps: ModalProps = {
  title: 'Demo',
  isOpen: true,
  onClose: () => {},
};
