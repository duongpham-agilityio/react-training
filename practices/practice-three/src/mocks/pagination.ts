import { PaginationProps } from '@/components';

export const defaultPropsPagination: PaginationProps = {
  data: [1, 2, 3, 4, 5, 6, 7],
  currentPage: 1,
  isNextPage: true,
  isPrevPage: false,
  onChangePage: () => {},
  onNextPage: () => {},
  onPreviousPage: () => {
    console.log(122);
  },
};
