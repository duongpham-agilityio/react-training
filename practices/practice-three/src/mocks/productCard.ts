import { ProductCardProps } from '@/components';

export const productCardProps: ProductCardProps = {
  info: {
    imageURL:
      'https://static.nike.com/a/images/t_PDP_1280_v1/f_auto,q_auto:eco/d455e5b6-23e9-4f06-8b86-32610d838937/renew-run-3-road-running-shoes-395S7c.png',
    title: 'Nike React Miler',
    description: 'Apple M1 Chip with 8‑Core CPU and 8‑Core GPU 512GB Storage',
    id: 1,
    price: 1000,
    status: false,
    isLike: false,
    statusMessage: 'Only 5 left',
  },
  onAddToCard: () => {},
  onLike: () => {},
};
