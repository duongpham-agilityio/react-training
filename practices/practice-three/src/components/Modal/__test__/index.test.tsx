import { fireEvent, render } from '@testing-library/react';

// Components
import ModalCustom from '..';

// Mocks
import { defaultModalProps } from '@/mocks';

const setup = (props = defaultModalProps) => render(<ModalCustom {...props} />);

describe('Modal', () => {
  it('Match to snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('Render with children', () => {
    const { getByText } = setup({
      ...defaultModalProps,
      children: <p>children</p>,
    });

    expect(getByText('children')).toBeDefined();
  });

  it('Close modal', () => {
    const onClose = jest.fn();
    const { getByRole } = setup({
      ...defaultModalProps,
      onClose,
    });

    fireEvent.click(getByRole('button'));

    expect(onClose).toBeCalled();
  });
});
