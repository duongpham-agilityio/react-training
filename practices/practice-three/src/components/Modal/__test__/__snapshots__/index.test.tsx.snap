// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Match to snapshot 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;
